src/PlusCommon/PlusCommon.txx

```diff
@@ -25,4 +27,22 @@ namespace PlusCommon
       return PLUS_SUCCESS;
     }
   }
+
+  //----------------------------------------------------------------------------
+  template<typename ElemType>
+  void JoinTokensIntoString(const std::vector<ElemType>& elems, std::string& output, char separator)
+  {
+    output = "";
+    std::stringstream ss;
+    typedef std::vector<ElemType> List;
+
+    for (List::const_iterator it = elems.begin(); it != elems.end(); ++it)
```

Comment:
Won't compile on clang. Need to add typename before List:

    for (typename List::const_iterator it = elems.begin(); it != elems.end(); ++it)